xxh_x86dispatch.c

```diff
@@ -53,8 +53,11 @@ extern "C" {
 #endif
 
 #if !(defined(__x86_64__) || defined(__i386__) || defined(_M_IX86) || defined(_M_X64))
-#  error "Dispatching is currently only supported on x86 and x86_64."
-#endif
+
+/* some build systems complain about empty units, so define this to save trouble */
+extern int __xxh_x86dispatch_non_empty_unit;
```

Comment:
👍

Reply:
It'd be nice to avoid `__` (double-underscore) prefix to follow the C standard clause ["Reserved identifiers"](https://stackoverflow.com/a/228797).

Reply:
triple-underscore ?

Reply:
Sorry for confusion.

1. Since xxHash always follow C standard, it'd be nice to avoid double-underscore prefix (e.g. `__xxh...`) to follow the C standard clause ["Reserved identifiers"](https://stackoverflow.com/a/228797).
   We also should avoid triple or further underscore prefix because it also contains double underscore prefix.  ( `strncmp("__", "___", 2) == 0` )

2. xxHash also cares basic C++ standard (not explicitly, but LZ4 has `ctocppcheck`).  xxHash has been avoiding underscore + upper case prefix (e.g. `_Xxh...`).

3. It's a not strict rule.  But conventionally and historically, all external xxHash macro, variable, function and enum names are started with `XXH`.  Some type name starts with `xxh`.

I think intention of this symbol is able to achieve without double underscore prefix.

Reply:
Yup, it's totally arbitrary, I can change it.

I'm just a bit confused - I thought the comment on #480 meant that approach was going to be pursued and this PR could be closed?

Reply:
@Cyan4973, could you suggest which PR (#477 or #480) should be pursued, abandoned or... mixed?

Reply:
Sorry, I will need a bit more time for this topic.

The short answer is : none will be pursued as is. 
I will probably borrow a bit of both, but mostly end up with a fairly different PR to support the requested feature.

Reply:
Keen to progress this. Anything I can do to help?

Reply:
Sorry, this is a difficult topic, that requires some dedicated focus.
And I'm sorry to report that I'm completely overwhelmed these days, with too many hot topics stealing my attention.

If you want to make progresses on this topic, you are absolutely welcome to do so.
It's one of the primary benefits of open source, that you can fork the project and make progresses on your own without further dependency from upstream. And even advocate for it as a better alternative for some use cases.
Hopefully, there will be a period in the future where it will be possible to focus and join efforts.